crates/runtime/src/lib.rs

```diff
@@ -124,7 +127,7 @@ impl<'a, T: ?Sized + Pointee> GuestPtr<'a, T> {
     }
 }
 
-impl<'a, T> GuestPtr<'a, [T]> {
+impl<'a, T: GuestType<'a>> GuestPtr<'a, [T]> {
```

Comment:
FWIW I'd recommend putting `where T: GuestType<'a>` on specific methods so you can still call methods like `offset_base` and `len` without having this trait bound.

Reply:
Good point. Also, one thing I forgot is that GuestType no longer provides a standalone validation function, its just part of read now. We'll have to either factor that back out for exposing a slice to work, or constrain `as_raw` to only work for types which accept any underlying bit pattern (i.e. not enums, unions, or flags)

Reply:
Introduced GuestTypeTransparent with the standalone validation func.